UpdateCosmosMessageView/handler.ts

```diff
@@ -0,0 +1,216 @@
+        status: {
+          archived: messageStatus.isArchived,
+          processing: messageStatus.status,
+          read: messageStatus.isRead
+        }
+      }
+    ),
+    TE.orElse(
+      flow(
+        TE.fromPredicate(
+          error =>
+            error.kind === "COSMOS_ERROR_RESPONSE" && error.error.code === 404,
+          identity
+        ),
+        // find and enrich message
+        TE.chain(() =>
+          messageModel.find([messageStatus.messageId, messageStatus.fiscalCode])
+        ),
+        TE.chainW(
+          TE.fromOption(
+            () =>
+              new Error(
+                `Message metadata not found for ${messageStatus.messageId}`
+              )
+          )
+        ),
+        TE.chainW(messageWithoutContent =>
+          pipe(
+            messageModel.getContentFromBlob(
+              blobService,
+              messageWithoutContent.id
+            ),
+            TE.chainW(
+              TE.fromOption(
+                () =>
+                  new Error(
+                    `Message body not found for ${messageWithoutContent.id}`
+                  )
+              )
+            ),
+            TE.map(content => ({ ...messageWithoutContent, content }))
+          )
+        ),
+        TE.map(messageWithContent => ({
+          components: {
+            attachments: {
+              has:
+                messageWithContent.content.legal_data?.has_attachment ?? false
+            },
+            euCovidCert: {
+              has: messageWithContent.content.eu_covid_cert !== null
+            },
+            legalData: {
+              has: messageWithContent.content.legal_data != null
+            },
+            payment: {
+              has: messageWithContent.content.payment_data != null
+            }
+          },
+          createdAt: messageWithContent.createdAt,
+          fiscalCode: messageWithContent.fiscalCode,
+          id: messageWithContent.id,
+          messageTitle: messageWithContent.content.subject,
+          senderServiceId: messageWithContent.senderServiceId,
+          status: {
+            archived: messageStatus.isArchived,
+            processing: messageStatus.status,
+            read: messageStatus.isRead
+          },
+          version: messageStatus.version
+        })),
+        // create message_view document
+        TE.chainEitherKW(MessageView.decode),
+        TE.chainW(messageView => messageViewModel.create(messageView))
+      )
+    )
+  );
+
+export const handle = (
+  telemetryClient: TelemetryClient,
+  messageViewModel: MessageViewModel,
+  messageModel: MessageModel,
+  queueClient: QueueClient,
+  blobService: BlobService,
+  rawMessageStatus: unknown
+): Promise<IStorableError<unknown> | MessageView> =>
+  pipe(
+    rawMessageStatus,
+    RetrievedMessageStatusWithFiscalCode.decode,
+    TE.fromEither,
+    TE.chain(handleStatusChange(messageViewModel, messageModel, blobService)),
+    TE.mapLeft(toStorableError(rawMessageStatus)),
```

Comment:
I think it should be explicit in the handle status change pipe so we can map directly failures, indicating for each if it's retriable or not and the related type

Reply:
in which case the retriable value could not be mapped from the error type?